### PR TITLE
add functionality for deactivate tariff for tree cases

### PR DESCRIPTION
**Pull request by LiliaMokhnatska**
# Green City UBS  Deactivaite tariff new cases 

## Summary Of Issue :
400 response in when diactivate tariff with parameters: 1) regionsId, stationsId; 2) citiesId, courierId, regionsId; 3) courierId, regionsId, stationsId 

It is expected that the 200 response is apear  

## Issue Link :
https://github.com/ita-social-projects/GreenCity/issues/5187

## Summary Of Changes :

1. add methods deactivateTariffsByRegionAndReceivingStations, deactivateTariffsByCourierAndRegionAndCities and deactivateTariffsByCourierAndRegionAndReceivingStations into DeactivateChosenEntityRepository.java
2. add methods shouldDeactivateTariffsByRegionAndReceivingStations, shouldDeactivateTariffsByCourierAndRegionAndCities, shouldDeactivateTariffsByCourierAndRegionAndReceivingStations into SuperAdminServiceImpl.java
3. change deactivateTariffForChosenParam method in SuperAdminServiceImpl.java
4. add tests for these cases in SuperAdminServiceImplTest.java

## How to test :
1. Admin need to be logged in http://localhost:8060/swagger-ui.html#/own-security-controller
2. Register Token 
3. Go to Main UBS menu -> Choose "Тарифи”
4. Click on tools “Gear”-> item “Деактивувати”.
5. Choose next parameters:
1) regionsId, stationsId;
or
2) citiesId, courierId, regionsId;
or
3) courierId, regionsId, stationsId

## CHECK LIST
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-#5058: Ticket title)
- [x] This template filled (above this section).
- [ ] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java [hunk 1]
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
             .forEach(tariffsInfo -> tariffsInfo.setLocationStatus(LocationStatus.DEACTIVATED));
     }
 
+    /**
+     * Method that deactivate tariffs for region id, list of cities id , list of
+     * station id and courier id.
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java]: Javadoc does not describe what the method does

File: dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java [hunk 2]
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
...
+     */
+    public void deactivateTariffsByRegionAndReceivingStations(Long regionId, List<Long> stationsId) {
+        entityManager.createQuery("select ti from TariffsInfo ti left join ti.receivingStationList rs "
+            + "left join ti.tariffLocations tl left join tl.location l"
+            + " where l.region.id =: regionId and "
+            + "rs.id in(:stationsId)", TariffsInfo.class)
+            .setParameter(REGION_ID, regionId)
+            .setParameter(STATIONS_ID, stationsId)
+            .getResultList()
+            .forEach(tariffsInfo -> tariffsInfo.setLocationStatus(LocationStatus.DEACTIVATED));
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java]: Can we use `UPDATE` here?

File: dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java [hunk 3]
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
...
+    }
+
+    /**
+     * Method that deactivate tariffs for region id, list of cities id and courier
+     * id.
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java]: please mode `id` to the previous line

File: dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java [hunk 4]
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
...
+            + "left join ti.tariffLocations tl  left join tl.location l"
+            + " where l.region.id =: regionId and "
+            + "tl.location.id in (:citiesId) and "
+            + "ti.courier.id =: courierId",
+            TariffsInfo.class)
+            .setParameter(REGION_ID, regionId)
+            .setParameter(CITIES_ID, citiesId)
+            .setParameter(COURIER_ID, courierId)
+            .getResultList()
+            .forEach(tariffsInfo -> tariffsInfo.setLocationStatus(LocationStatus.DEACTIVATED));
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java]: same here

File: dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java [hunk 5]
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
...
+    }
+
+    /**
+     * Method that deactivate tariffs for region id, list of station id and courier
+     * id.
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java]: same here

File: dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java [hunk 6]
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
...
+        entityManager.createQuery("select ti from TariffsInfo ti left join ti.receivingStationList rs "
+            + "left join ti.tariffLocations tl  left join tl.location l"
+            + " where l.region.id =: regionId and "
+            + "rs.id in(:stationsId) and ti.courier.id =: courierId",
+            TariffsInfo.class)
+            .setParameter(REGION_ID, regionId)
+            .setParameter(STATIONS_ID, stationsId)
+            .setParameter(COURIER_ID, courierId)
+            .getResultList()
+            .forEach(tariffsInfo -> tariffsInfo.setLocationStatus(LocationStatus.DEACTIVATED));
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java]: same here

File: service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java [hunk 7]
```diff
@@ -96,6 +96,13 @@ public class SuperAdminServiceImpl implements SuperAdminService {
     private static final String RECEIVING_STATIONS_OR_COURIER_EXIST_MESSAGE =
         "Current receiving stations: %s or courier: %s don't exist.";
     private static final String REGION_OR_COURIER_EXIST_MESSAGE = "Current region: %s or courier: %s don't exist.";
+    private static final String REGION_OR_RECEIVING_STATIONS_EXIST_MESSAGE =
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java]: I see in the constant name `STATIONS_EXIST` and in the message that `stations: %s don't exist`

File: service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java [hunk 8]
```diff
@@ -96,6 +96,13 @@ public class SuperAdminServiceImpl implements SuperAdminService {
...
+        "Current region: %s or receiving stations: %s don't exist.";
+    private static final String REGION_OR_CITIES_OR_COURIER_EXIST_MESSAGE =
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java]: same here

File: service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java [hunk 9]
```diff
@@ -96,6 +96,13 @@ public class SuperAdminServiceImpl implements SuperAdminService {
...
+        "Current region: %s or cities: %s or courier: %s don't exist.";
+
+    private static final String REGION_OR_RECEIVING_STATIONS_OR_COURIER_EXIST_MESSAGE =
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java]: same here

File: dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java (hunk 2 above)
Comment (LiliaMokhnatska) [reply dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java]: Ok. I'll try to use update.

File: service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java [hunk 10]
```diff
@@ -858,6 +874,101 @@ private boolean shouldDeactivateTariffsByCourierAndReceivingStations(DetailsOfDe
...
+            } else {
+                throw new BadRequestException(BAD_SIZE_OF_REGIONS_MESSAGE);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Method that checks if the tariff should be deactivated by courier id, regions
+     * id and receiving stations id. In this case size of RegionsList should be one
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java]: same here

File: service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java [hunk 11]
```diff
@@ -858,6 +874,101 @@ private boolean shouldDeactivateTariffsByCourierAndReceivingStations(DetailsOfDe
...
+            } else {
+                throw new BadRequestException(BAD_SIZE_OF_REGIONS_MESSAGE);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Method that checks if the tariff should be deactivated by courier id, regions
+     * id and cities. In this case size of RegionsList should be one because we
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java]: move `id` to the previous line

**Review by RomaMocherniuk — APPROVED** [review]

File: service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java (hunk 7 above)
Comment (LiliaMokhnatska) [reply service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java]: Yes, I change it

File: dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java (hunk 1 above)
Comment (LiliaMokhnatska) [reply dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java]: changed

**Review by ABbondar — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Review by KharchenkoV — APPROVED** [review]

**Pull request merged**